Content/default/src/Client/Index.fs

```diff
@@ -1,43 +1,46 @@
 module Index
 
 open Elmish
-open Fable.Remoting.Client
+open SAFE
 open Shared
 
-type Model = { Todos: Todo list; Input: string }
+type Model = {
+    Todos: Deferred<Todo list>
+    Input: string
+}
 
 type Msg =
-    | GotTodos of Todo list
     | SetInput of string
-    | AddTodo
-    | AddedTodo of Todo
+    | GetTodos of AsyncOperation<unit, Todo list>
+    | AddTodo of AsyncOperation<unit, Todo>
 
-let todosApi =
-    Remoting.createApi ()
-    |> Remoting.withRouteBuilder Route.builder
-    |> Remoting.buildProxy<ITodosApi>
+let todosApi = Api.makeProxy<ITodosApi> ()
 
 let init () =
-    let model = { Todos = []; Input = "" }
-    let cmd = Cmd.OfAsync.perform todosApi.getTodos () GotTodos
-    model, cmd
+    let model = { Todos = NotStarted; Input = "" }
+    model, Cmd.ofMsg (GetTodos(Start()))
 
 let update msg model =
     match msg with
-    | GotTodos todos -> { model with Todos = todos }, Cmd.none
     | SetInput value -> { model with Input = value }, Cmd.none
-    | AddTodo ->
-        let todo = Todo.create model.Input
-
-        let cmd = Cmd.OfAsync.perform todosApi.addTodo todo AddedTodo
-
-        { model with Input = "" }, cmd
-    | AddedTodo todo ->
-        {
-            model with
-                Todos = model.Todos @ [ todo ]
-        },
-        Cmd.none
+    | GetTodos msg ->
+        match msg with
+        | Start() ->
+            let cmd = Cmd.OfAsync.perform todosApi.getTodos () (Finished >> GetTodos)
+            { model with Todos = InProgress }, cmd
+        | Finished todos -> { model with Todos = Resolved todos }, Cmd.none
+    | AddTodo msg ->
+        match msg with
+        | Start() ->
+            let todo = Todo.create model.Input
+            let cmd = Cmd.OfAsync.perform todosApi.addTodo todo (Finished >> AddTodo)
+            { model with Input = "" }, cmd
+        | Finished todo ->
+            {
+                model with
+                    Todos = model.Todos.Map(fun todos -> todos @ [ todo ])
```

Comment:
It seems weird to me for `Map` to be a method rather than a function in the `Deferred` module.

Reply:
It's also a function in that module - I just left them as both functions and members. Happy to remove the members though!

Reply:
Ditto above.